### DISCUSSION TITLE
Force current directory to be the workspaceFolder

### DISCUSSION THREAD

**Discussion by amgodoi** [neoclide/coc.nvim#4466] [Q&A]
Is there any way to force current directory to be the workspaceFolder?

I do not want any pattern to confirm the workspaceFolder ... I just want the current directory.

I read https://github.com/neoclide/coc.nvim/wiki/Using-workspaceFolders, but unsuccessful.

Is there any way?

**Comment by fannheyward** [comment]
I don't think coc has this feature by now.

**Reply by amgodoi** [reply]
Hi!

From the wiki:

"When resolve of workspace folder failed, cwd is used for workspaceFolder, ..."

Sometimes I just want to test something without any project and use a directory without any pattern. It's supposed to be cwd as workspace folder (as said in wiki), but an "old project" is considered as workspace folder.

Appears a message like this:
"Main.java is not on the classpath of project XXX".

After that coc-nvim does not work properly.

Could it be a bug?

**Comment by chemzqm** [comment]
You can use `:h g:WorkspaceFolders` before coc.nvim service start.

**Reply by amgodoi** [reply]
I tried this:

autocmd FileType java let b:coc_root_patterns = ['pom.xml', '.classpath']

Then I create an empty .classpath or pom.xml as necessary ... just to resolve the workspace folder.

Solved. Thanks.
